plugins/DebuggerCore/unix/linux/DebuggerCore.cpp

```diff
@@ -553,8 +553,10 @@ std::shared_ptr<IDebugEvent> DebuggerCore::handle_event(edb::tid_t tid, int stat
 		const auto signo = WSTOPSIG(status);
 		if(signo == SIGILL || signo == SIGSEGV) {
 			// no need to peekuser for SIGILL, but have to for SIGSEGV
+			State state;
+			(*it)->get_state(&state);
```

Comment:
This will fetch the whole state even when we don't need to (see the comment above).